doc/tutorial/single_cell_recipe.rst

```diff
@@ -0,0 +1,181 @@
+    cell = arbor.cable_cell(tree, labels, decor)
+
+The recipe
+----------
+
+The :class:`arbor.single_cell_model` of the previous example created a :class:`arbor.recipe` under
+the hood, and abstracted away the details so we were unaware of its existence.
+
+Creating an analogous recipe starts with creating a class that inherits from :class:`arbor.recipe` and overrides and implements some of :class:`arbor.recipe` methods. Not all methods have to be overridden, but some will always have to be, such as :meth:`arbor.recipe.num_cells`. It returns `0` by default and models without cells are quite boring!
+
+.. code-block:: python
+
+    # (4) Define a recipe for a single cell and set of probes upon it.
+    # This constitutes the corresponding generic recipe version of
+    # `single_cell_model.py`.
+
+    class single_recipe (arbor.recipe):
+        def __init__(self, cell, probes):
+            # (4.1) The base C++ class constructor must be called first, to ensure that
+            # all memory in the C++ class is initialized correctly.
+            arbor.recipe.__init__(self)
+            self.the_cell = cell
+            self.the_probes = probes
+            self.the_props = arbor.neuron_cable_propetries()
+            self.the_cat = arbor.default_catalogue()
+            self.the_props.register(self.the_cat)
+
+        def num_cells(self):
+            # (4.2) Override the num_cells method
+            return 1
+
+        def num_sources(self, gid):
+            # (4.3) Override the num_sources method
+            return 1
+
+        def cell_kind(self, gid):
+            # (4.4) Override the cell_kind method
+            return arbor.cell_kind.cable
+
+        def cell_description(self, gid):
+            # (4.5) Override the cell_description method
+            return self.the_cell
+
+        def probes(self, gid):
+            # (4.6) Override the probes method
+            return self.the_probes
+
+        def global_properties(self, kind):
+            # (4.7) Override the global_properties method
+            return self.the_props
+
+Step **(4)** describes the recipe that will reflect our single cell model.
+
+Step **(4.1)** defines the class constructor. It can take any shape you need, but it is important to call base class' constructor. In general it's a good idea to make objects that'll be return by methods that need to be overridden members of the class. With this constructor, we could easily change the cell and probes of the model, should be want to do so. Here we initialize the cell properties to match Neuron's defaults using Arbor's built-in :meth:`arbor.neuron_cable_properties` and extend with Arbor's own :meth:`arbor.default_catalogue`.
+
+Step **(4.2)** defines that this model has one cell.
+
+Step **(4.3)** defines that this model has one source.
+
+Step **(4.4)** returns the :class:`arbor.cell_kind.cable`. Make sure you don't instantiate the model with a non cable cell! Since we have just one cell, we can ignore :gen:`gid`.
```

Comment:
returns :class:`arbor.cell_kind.cable` (minus the _the_). 
I think instead of cautioning users against using a non-cable cell, we should explain that the reason we are returning `arbor.cell_kind.cable` is because our modeled cell is of that kind.

Reply:
I thought about this (second comment). I thought that is obvious enough, but not that nothing will prevent you from constructing this recipe with non-cable cell `cell`, so hence the warning. It would be nice to query a `cell` for its `cell_kind` and set or test in the constructor.

Reply:
It is the responsibility of the user to not mess this up. And it's not just the `cell_kind`, it's also the `num_sources` and `num_targets` that need to match the cell description. 
We can keep the comment as is, I just think the logic is reversed: we don't build a cable cell because the kind of the cell is `cable`; the kind of the cell is `cable` because we're building a cable cell.

Reply:
> It is the responsibility of the user to not mess this up.

I'm not happy with this line of reasoning. Python is typed (typing was invented to help users not make mistakes), so we should warn users when we deviate and suddenly request users to repeat themselves.

Reply:
It may be asking too much of the user, and I support having a "recipe checker" of some sort, but it should be, yet again, up to the user to check the recipe. In any case, we do warn the user, in the form of exceptions at runtime.

As to this actual comment, I still think the logic is reversed. Warning not to use a something other than a cable cell places the importance on the cell kind: it has a cable kind, therefore we create must create a cable cell, when it should be the other way around. I would put the `cell_description` right beneath `num_cells` in the recipe, and then when describing the `cell_kind` I would refer back to the `cell_description` saying that the kind needs to be `cable` because the actual cell is a cable cell.

Reply:
In any case, I don't have any major objections to the comment itself. The recipe concept can be complicated to describe to the users, I think making it so that `cell_description` is the main method that returns the cell, and `cell_kind`, `num_sources` and `num_targets` provide easy access to information about the cell, without actually building the cell by calling `cell_descrition`, could make the concept easier to understand, potentially.

Reply:
One person interested in Arbor wanted to simulate one or some cells modelled as a cable cell in a large network of LIF cells. So use cases with multiple kinds might not be rare.

I have reworded the warning.